### DISCUSSION TITLE
[@types/node] node:stream/promises pipeline() options parameter is missing "end" property

### DISCUSSION THREAD

**Discussion by NemoStein** [DefinitelyTyped/DefinitelyTyped#63579] [Issues with a @types package]
The `node:stream/promises` `pipeline()` function accepts an options object as last parameter, this object accepts 2 optional properties and should be declared as:
```ts
interface PipelineOptions {
  end?: boolean
  signal?: AbortSignal
}
```
But the `@types/node` definition only includes the mandatory `signal` property:
```ts
interface PipelineOptions {
  signal: AbortSignal
}
```
Although undocumented, the `end` property was added/merged in January, 11th (https://github.com/nodejs/node/issues/34805#issuecomment-1009739825).
I've already opened a issue to track the nodejs docs update (https://github.com/nodejs/node/issues/45821), but I don't see any reason to not update the types already.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
depends on: nodejs/node#45832

**Comment by leftshift** [comment]
Any news on this? Also just ran into this issue.

**Comment by xfournet** [comment]
PR submitted https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65212
